worker/src/main/java/gov/cms/ab2d/worker/service/JobProcessingServiceImpl.java

```diff
@@ -146,6 +158,18 @@ public Job processJob(final String jobId) {
         return jobOutputs;
     }
 
+    private boolean isOptOutPatient(String patientId) {
+
+        final List<Consent> consents = consentRepository.findByHicn(patientId);
+        if (consents.isEmpty()) {
+            // No opt-out record found for this patient - Opt-In by default.
+            return false;
+        }
+
+        // found at least one opt-out record for this patient. This patient has opted-out
+        return true;
```

Comment:
Opt-out _could_ be effective on or after a certain date. So let's get the date from the `Consent` record, and if it is in the _future_, consider the Patient still _opted in_.

Reply:
done